packages/server/graphql/private/mutations/stripeFailPayment.ts

```diff
@@ -56,10 +57,19 @@ const stripeFailPayment: MutationResolvers['stripeFailPayment'] = async (
   }
   const {creditCard, stripeSubscriptionId} = org
 
-  if (paid || stripeSubscriptionId !== subscription) return {orgId}
+  if (paid || stripeSubscriptionId !== subscription || stripeSubscriptionId === null) return {orgId}
 
   // RESOLUTION
-  await terminateSubscription(orgId)
+  const subscriptionObject = await manager.retrieveSubscription(stripeSubscriptionId)
+
+  if (subscriptionObject.status === 'incomplete' || subscriptionObject.status === 'canceled') {
+    // Terminate subscription if it is the first failed payment, or it is already canceled
```

Comment:
Why terminate the subscription on **1st** failed payment?

Reply:
@tianrunhe I keep what is happening today only for case 1: 
User setup the subscription for the first time, he adds a card but then almost immediately first payment fails.

I keep this for two reasons:
- I think it is safe to cancel subscription in this case.
- Stripe [documentation](https://stripe.com/docs/billing/subscriptions/overview#payment-window) suggest to cancel such subscriptions if no payment was made within 23 hours. 
To not handle this 23 hours case specifically I think it is easier to just cancel right now:

```
Payment window
Customers have about 23 hours to make a successful payment. The subscription remains in status incomplete and the invoice is open during this time. If your customer pays the invoice, the subscription updates to active and the invoice to paid. If they don't make a payment, the subscription updates to incomplete_expired and the invoice becomes void.

This window exists because your customer usually makes the first payment for a subscription while on-session. If the customer returns to your application after 23 hours, create a new subscription for them.
```

Reply:
Added more comments in code. What do you think?